src/main/java/org/tron/core/db/ByteArrayWrapper.java

```diff
@@ -45,7 +45,7 @@ public ByteArrayWrapper(byte[] data) {
    * equals Objects.
    */
   public boolean equals(Object other) {
-    if (!(other instanceof ByteArrayWrapper)) {
+    if (this.getClass() != other.getClass()) {

```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "other" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=853&issues=AWOg829SmXhJxGlVU5lM&open=AWOg829SmXhJxGlVU5lM)

[ISSUE_KEY]: AWOg829SmXhJxGlVU5lM